kiro: init at 0.2.13

Vuks69 opened the pull request:
Functioning package. Login works, editor and LLM integration works.
resolves kirodotdev/Kiro/issues/169
duplicates #425565 - ~~@deftdawg, feel free to use this PR as you see fit as I don't really have the intention of maintaining this at the time (created this for personal use and saw it wasn't yet on nixpkgs, before you posted your PR)~~ this PR is used instead of the mentioned one.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Vuks69 committed:
kiro: init at 0.2.13

deftdawg commented:
Was able to login with one of the previous commits on this PR about an hour ago, will try to find to play with the latest later. Good stuff @Vuks69 🚀

amadaluzia reviewed (changes requested):
Welcome to nixpkgs! 

It is a malpractice to commit a package but not maintain it. If you can, please add yourself as a maintainer and mark yourself as a maintainer for the package.

It also seems as if you can install Kiro on macOS, and Linux ARM I am guessing since I can't check myself. It would be nice if you could add platform support for that.

midischwarz12 reviewed (changes requested):
Please make a commit to add yourself to the nixpkgs maintainer list as described in the [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Vuks69 commented on pkgs/by-name/ki/kiro/package.nix:
```diff
@@ -0,0 +1,60 @@
+  inherit useVSCodeRipgrep;
+  commandLineArgs = extraCommandLineArgs;
+
+  version = "0.1.0";
```
This is what the `kiro` binary returns when you run `kiro --version`, so I decided to use that instead. I'll ask in their discord what should be put there.

midischwarz12 replied:
Oh really? I couldn't find it anywhere in their codebase. Usually, we prefer to mark the version of a tagged release. If you reach out, maybe make the tag suggestion.

Vuks69 replied:
Jugding by discussions in their discord, that version is being updated. Seen a mention of 0.1.6 somewhere.

midischwarz12 replied:
Okay, are we sure that `0.1.0` is the currect version for this package then? Asking b/c 0.1.6 > 0.1.0

midischwarz12 replied:
Yea, that sounds like a pain and a half. Good luck.

Vuks69 replied:
welp, with the linux tarball it's simple enough:
`jq .version Kiro/resources/app/package.json`

No idea how to deal with the MacOS .dmg packages though. Don't have a Mac.

Vuks69 commented on pkgs/by-name/ki/kiro/package.nix:
```diff
@@ -0,0 +1,60 @@
+    license = lib.licenses.unfree;
+    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+    maintainers = with lib.maintainers; [ ];
+    platforms = [ "x86_64-linux" ];
```
Right now I'm unable to determine if Kiro uses the same tarball for both linux and darwin, because they have locked access to the actual download links behind a waitlist due to demand.

If anyone has a download link for MacOS specifically, please let me know what it looks like and whether it shares the build number/ID with the linux one.

deftdawg replied:
Happened to have the download page open:
MacOS Apple Silicon: https://prod.download.desktop.kiro.dev/releases/202507161958-Kiro-dmg-darwin-arm64.dmg
MacOS Intel: https://prod.download.desktop.kiro.dev/releases/202507162010-Kiro-dmg-darwin-x64.dmg

amadaluzia replied:
Happy to mark as resolved?

Vuks69 commented on pkgs/by-name/ki/kiro/package.nix:
```diff
@@ -0,0 +1,60 @@
+    homepage = "https://kiro.dev";
+    license = lib.licenses.unfree;
+    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+    maintainers = with lib.maintainers; [ ];
```
Does the commit to the maintainers list have to be done in a separate PR, or can it be done within this one?

midischwarz12 commented on pkgs/by-name/ki/kiro/package.nix:
```diff
@@ -0,0 +1,60 @@
+  updateScript = null;
+
+  meta = {
+    description = "The AI IDE for prototype to production built on VSCode";
```
I agree with this. It's better wording.

deftdawg replied:
An IDE for Agentic AI workflows based on VS Code

deftdawg commented:
> I don't think this package can be merged into nixpkgs considering it seemingly abuses an oversight by the copyright holders and bypasses the waitlist. If this gets patched, then we have a source invalidation.

Given the issue on their GitHub issues was triaged as a feature and there was no waitlist when this PR was opened yesterday, I'm inclined to believe the authors do want their software distributed.

Are we sure this bypasses the waitlist? Did you try it? I can't test it because I registered yesterday before there was a wait list, when opening the app it required me to complete an oauth registration prior to using it, that's actually where my PR broke (wouldn't call browser). If the authors aren't blocking new registrations at the OAuth registration step, the waitlist will be bypassed by anyone determined in very short order.

normalcea commented:
> required me to complete an oauth registration prior to using it

Oh I understand this better now, I'm guessing the license agreement/copy protection mechanism happens on startup. I don't have any objections then.

Vuks69 commented:
I added most of the changes suggested, but honestly this'll have to be put on the backburner until Kiro devs give us a simple/reasonable way to get the tarballs/dmg files **with matching versions**. Until then, right now I have a 0.1.0 version for x64 linux, and 0.1.9 for x64/arm darwin. I don't see a way to manually figure out the download links as they include the build timestamp, I'm not going to abuse the endpoint by bruteforcing that.

I whipped up a small script that utilizes dmg2img and apfs-fuse to read the .dmg's, resulting in this:
```
❯ /home/vuks/src/nixpkgs/pkgs/by-name/ki/kiro/update.sh

File: 202507140012-distro-linux-x64.tar.gz
Version: 0.1.0
Hash: sha256-6bbc/HndiN/HUoZyYo9r6Olih2n4/NMyRpxD59z9SH0=

File: 202507162010-Kiro-dmg-darwin-x64.dmg
Version: 0.1.9
Hash: sha256-qpLIZC93AeyBVnEZM8Jf4EOioS/deCJjPS2cCk2q5jo=

File: 202507161958-Kiro-dmg-darwin-arm64.dmg
Version: 0.1.9
Hash: sha256-+QfvhZ+VRnylR0Pj9Vjc43cqcR8lzh0oqAEW9FNTdCk=
```

Vuks69 commented:
Actually I think I managed to find the link they're using for their internal update process, in the package.json file. This might allow me to grab the latest artifacts, including the ARM one for linux. Stay tuned.

Vuks69 commented:
Unless I can find download links for anything else than x86_64-linux that are on version 0.1.0, I'm giving up on including anything other than that for now.
While I can find the update links for linux, they seem to still be marked as 0.1.0 (despite latest as of now is 0.1.15), the links would point to a version that I'm unable to determine the version of, and whether they're the full tarball or some sort of diff/incremental tarball.
I'm entirely unable to figure out the update links for darwin as mentioned earlier. At this point I'd rather just have a working MVP out there and wait until Kiro devs give packagers a way to properly determine the download sources.

Vuks69 committed:
maintainers: add vuks

Vuks69 commented on pkgs/by-name/ki/kiro/package.nix:
```diff
@@ -0,0 +1,60 @@
+
+let
+  build = "202507140012";
+  sources = {
```
In the name of KISS/not reinventing the wheel I'd rather keep it the way it is, similar to how Cursor does it, unless there's serious benefits to it.

amadaluzia replied:
Happy to mark this as resolved?

Vuks69 commented on pkgs/by-name/ki/kiro/package.nix:
```diff
@@ -0,0 +1,60 @@
+  vscode-generic,
+  fetchurl,
+  extraCommandLineArgs ? "",
+  useVSCodeRipgrep ? stdenv.hostPlatform.isDarwin,
```
Since it's confirmed that darwin packages exist, I'll leave this in so I don't have to change it again later when I figure out the download links for them.

Vuks69 replied:
It does exist and I'm able to find source for darwin packages.

deftdawg commented:
There is a site that has a list of links to the releases...

https://kirohistory.dev/

It includes this which is supposed  to be 0.1.15:
https://prod.download.desktop.kiro.dev/releases/202507180237--distro-linux-x64-tar-gz/202507180237-distro-linux-x64.tar.gz

(I'm on my phone so can't test it)

deftdawg commented:
Confirmed it's the newer build:
Version: 0.1.15
VSCode Version: 1.94.0
Commit: 3bc33dd4c5eafbd5052855ecbf5c622c75d33ab8
Date: 2025-07-18T01:48:26.344Z
Electron: 30.5.1
ElectronBuildId: undefined
Chromium: 124.0.6367.243
Node.js: 20.16.0
V8: 12.4.254.20-electron.0
OS: Linux x64 6.12.35

Here's the patch to your branch @Vuks69 
```patch
diff --git a/pkgs/by-name/ki/kiro/package.nix b/pkgs/by-name/ki/kiro/package.nix
index 50adaaa71..c9f20a6ce 100644
--- a/pkgs/by-name/ki/kiro/package.nix
+++ b/pkgs/by-name/ki/kiro/package.nix
@@ -10,11 +10,11 @@

 let
   inherit (stdenv) hostPlatform;
-  version = "0.1.0";
+  version = "0.1.15";
   sources = {
     x86_64-linux = fetchurl {
-      url = "https://prod.download.desktop.kiro.dev/releases/202507140012--distro-linux-x64-tar-gz/202507140012-distro-linux-x64.tar.gz";
-      hash = "sha256-6bbc/HndiN/HUoZyYo9r6Olih2n4/NMyRpxD59z9SH0=";
+      url = "https://prod.download.desktop.kiro.dev/releases/202507180237--distro-linux-x64-tar-gz/202507180237-distro-linux-x64.tar.gz";
+      hash = "sha256-cXDG0WBKEZR3e7hrTqVKRy3diTvwicy5RPi+qTkPbwc=";
     };
   };
   src = sources.${hostPlatform.system} or (throw "Unsupported platform for Kiro");
```

Vuks69 commented:
oh that is super useful and clean, in browser when hovering over the download links I can see they just link straight to the official download links. that lets me put the other archs and darwin in the package too <3

amadaluzia reviewed (approved):
LGTM, make sure to change the name of the pull request to match the name of your commit for it to get verified by GitHub.

4reblue commented:
Just a quick reminder: Using an alternative download link lets users bypass the waitlist and access Kiro, even with a new account. I think the Kiro team prefers to keep this quiet and removed the download link from the official site, as it could increase demand and delay sending invites to those on the waitlist.

<img width="541" height="130" alt="image" src="https://github.com/user-attachments/assets/8fc52dfb-c38b-4e94-9c99-734fa68c8aa7" />

deftdawg commented:
> Just a quick reminder: Using an alternative download link lets users bypass the waitlist and access Kiro, even with a new account. I think the Kiro team prefers to keep this quiet since they removed the download link from the official site, as it could increase demand and delay sending invites to those on the waitlist.

See: https://github.com/NixOS/nixpkgs/pull/425607#issuecomment-3086073643

Only place Kiro(Amazon) can implement a wait list is during the OAuth registration step that new users must complete after first opening Kiro.  Can you personally confirm they haven't been able to stop new registrations at that step? (I can't my account was registered prior to the waitlist stuff)

Kiro team could delete the downloads or put some authorization requirements on the download links (which are widely known and probably mirrored)... which they also haven't done.

deftdawg commented:
OAuth registration for new users goes to wait list... 

Here's the patch to @Vuks69's branch to update to 0.1.25
```diff
diff --git a/pkgs/by-name/ki/kiro/package.nix b/pkgs/by-name/ki/kiro/package.nix
index 628c44634..05d84799f 100644
--- a/pkgs/by-name/ki/kiro/package.nix
+++ b/pkgs/by-name/ki/kiro/package.nix
@@ -10,20 +10,20 @@
 
 let
   inherit (stdenv) hostPlatform;
-  version = "0.1.15";
+  version = "0.1.25";
   sources = {
     x86_64-linux = fetchurl {
-      url = "https://prod.download.desktop.kiro.dev/releases/202507180237--distro-linux-x64-tar-gz/202507180237-distro-linux-x64.tar.gz";
-      hash = "sha256-cXDG0WBKEZR3e7hrTqVKRy3diTvwicy5RPi+qTkPbwc=";
+      url = "https://prod.download.desktop.kiro.dev/releases/202507232027--distro-linux-x64-tar-gz/202507232027-distro-linux-x64.tar.gz";
+      hash = "sha256-XxtJS7hLUKHJG0HhiA2Fyt7wNh5YebOOYadvNvXCKzc=";
     };
     aarch64-linux = sources.x86_64-linux; # "universal" package
     x86_64-darwin = fetchurl {
-      url = "https://prod.download.desktop.kiro.dev/releases/202507180243-Kiro-dmg-darwin-x64.dmg";
-      hash = "sha256-dXOy8J8rF0nFHjY+swy5JXDo5m1dkZgXsENN9f/Nx9o=";
+      url = "https://prod.download.desktop.kiro.dev/releases/202507232041-Kiro-dmg-darwin-x64.dmg";
+      hash = "sha256-dH+aQ8IKqGC0TLMlOQFk7UYhL5/dWSOwLXD417bBpWk=";
     };
     aarch64-darwin = fetchurl {
-      url = "https://prod.download.desktop.kiro.dev/releases/202507180224-Kiro-dmg-darwin-arm64.dmg";
-      hash = "sha256-k4E3VPhdZJziA47TGMfoDcOO4kl3ORlRMW9kiHxmUJ8=";
+      url = "https://prod.download.desktop.kiro.dev/releases/202507232015-Kiro-dmg-darwin-arm64.dmg";
+      hash = "sha256-GQ40ZirD+MEHcS6B7i7xl0wYCPspvokRrMuD3eqZQM4=";
     };
   };
   src = sources.${hostPlatform.system} or (throw "Unsupported platform for Kiro");
```

Vuks69 commented:
Thanks for the patch! Applied it and updated commitmsg and PR title.

normalcea reviewed (changes requested):
I am unable to run on my Macbook M1 ("aarch64-linux" via Asahi). VSCodium executes correctly so this isn't an asahi quirk on my end. 

```
/nix/store/f914arnv5z40glfrn0v5xrd7g8zd9597-kiro-0.1.25/bin/.kiro-wrapped: line 63: /nix/store/f914arnv5z40glfrn0v5xrd7g8zd9597-kiro-0.1.25/lib/kiro/kiro: cannot execute binary file: Exec format error
```

```
aarch64-linux = sources.x86_64-linux; # "universal" package
```

I also don't understand what this means. If aarch64-linux isn't supported or we aren't confident that it is, then it shouldn't be listed. Same goes for darwin as well.

Vuks69 commented:
That's unfortunate - Kiro distributes the linux tarball as a "universal" package, which I assumed meant it's for all architectures. Turns out they most likely meant it's for all distributions (given the other linux package is a .deb file). The darwin packages should work correctly, as they're explicitly for x64 and Apple ARM.
I'll remove the linux arm package and ask on their discord about linux arm support.

deftdawg commented:
0.1.42 is out, I also had kiro make itself an updater script, not sure the best way to share though.
```diff
--- pkgs/by-name/ki/kiro/package.nix.backup     2025-08-05 12:23:59.475847100 -0400
+++ pkgs/by-name/ki/kiro/package.nix    2025-08-05 12:28:15.840931943 -0400
@@ -10,20 +10,20 @@
 
 let
   inherit (stdenv) hostPlatform;
-  version = "0.1.25";
+  version = "0.1.42";
   sources = {
     x86_64-linux = fetchurl {
-      url = "https://prod.download.desktop.kiro.dev/releases/202507232027--distro-linux-x64-tar-gz/202507232027-distro-linux-x64.tar.gz";
-      hash = "sha256-XxtJS7hLUKHJG0HhiA2Fyt7wNh5YebOOYadvNvXCKzc=";
+      url = "https://prod.download.desktop.kiro.dev/releases/202508020245--distro-linux-x64-tar-gz/202508020245-distro-linux-x64.tar.gz";
+      hash = "sha256-2kUx4PEqVcLF2/gk1+bxjeVrfX6wOxfRpLeMevNNsR8=";
     };
     aarch64-linux = sources.x86_64-linux; # "universal" package
     x86_64-darwin = fetchurl {
-      url = "https://prod.download.desktop.kiro.dev/releases/202507232041-Kiro-dmg-darwin-x64.dmg";
-      hash = "sha256-dH+aQ8IKqGC0TLMlOQFk7UYhL5/dWSOwLXD417bBpWk=";
+      url = "https://prod.download.desktop.kiro.dev/releases/202508020251-Kiro-dmg-darwin-x64.dmg";
+      hash = "sha256-8P562N6b4PyNYgNICyrlVxphkQjXdv80VEXtk71iabE=";
     };
     aarch64-darwin = fetchurl {
-      url = "https://prod.download.desktop.kiro.dev/releases/202507232015-Kiro-dmg-darwin-arm64.dmg";
-      hash = "sha256-GQ40ZirD+MEHcS6B7i7xl0wYCPspvokRrMuD3eqZQM4=";
+      url = "https://prod.download.desktop.kiro.dev/releases/202508020230-Kiro-dmg-darwin-arm64.dmg";
+      hash = "sha256-YBHnFmoVfrl4chEkj9YBP2aUZ4g9UY9AixgCFKV5FVU=";
     };
   };
   src = sources.${hostPlatform.system} or (throw "Unsupported platform for Kiro");
```

deftdawg commented:
It works standalone, but I wasn't able to test it with nixpkgs-update (I think because kiro doesn't exist in master, but not certain).

[update.sh.txt](https://github.com/user-attachments/files/21604406/update.sh.txt)

```diff
--- a/pkgs/by-name/ki/kiro/package.nix
+++ b/pkgs/by-name/ki/kiro/package.nix
@@ -49,7 +49,7 @@ in
   patchVSCodePath = true;

   tests = { };
-  updateScript = null;
+  updateScript = ./update.sh;

   meta = {
     description = "IDE for Agentic AI workflows based on VS Code";
```

Vuks69 commented:
@deftdawg - thanks for the heads up and the update script - I let myself beautify it a little.
@normalcea - I think I've addressed all of your concerns now, let me know if anything's still off.

Tested the update script and building/running Kiro on x64 linux.

normalcea commented:
### Review 2

**issue** (blocking): `updateScript` fails due to non-portable shebang.

Using  `nix-shell maintainers/scripts/update.nix --argstr skip-prompt true --argstr package kiro` throws an error as dependencies are not handled in a nix portable way. We need to use a nix-shell instead.

**Note**: The command above is how your update scripts should be tested as it's what is used for both maintainers and CI.

Diff for resolving (you can look at other `update.sh` scripts in-tree for more examples)

``` diff
diff --git a/pkgs/by-name/ki/kiro/update.sh b/pkgs/by-name/ki/kiro/update.sh
index b99a0bb284ee..3d35aefdb071 100755
--- a/pkgs/by-name/ki/kiro/update.sh
+++ b/pkgs/by-name/ki/kiro/update.sh
@@ -1,4 +1,5 @@
-#!/usr/bin/env bash
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl gnugrep gnused jq

 set -euo pipefail

@@ -25,16 +26,6 @@ error_exit() {
     exit 1
 }

-# Check dependencies
-check_dependencies() {
-    local deps=("curl" "jq" "nix")
-    for dep in "${deps[@]}"; do
-        if ! command -v "$dep" >/dev/null 2>&1; then
-            error_exit "Required dependency '$dep' not found"
-        fi
-    done
-}
-
 # Fetch metadata for a platform
 fetch_platform_info() {
     local url="$1"
@@ -121,9 +112,6 @@ generate_sources_json() {
 main() {
     echo "Starting Kiro update process..."

-    # Check dependencies
-    check_dependencies
-
     # Fetch metadata for all platforms
     echo "Fetching platform information..."
     for platform in "${!PLATFORM_URLS[@]}"; do

```

normalcea reviewed (changes requested):
See review 2

JohnRTitor commented:
Let me know when this is ready, I'll take a final look and merge.

JohnRTitor reviewed (approved):
Tested the update script and the package

(I don't have access code, so I couldn't test the IDE)

But LGTM, runs without a hitch
